src/lib/prov/pkcs11/p11.h

```diff
@@ -24,7 +24,14 @@
    #define CK_DECLARE_FUNCTION(returnType, name) returnType name
 #endif
 
-#if defined(_MSC_VER)
+/* When building with clang-cl, specifying this with dllimport causes a warning when defining the
+ * CK_C_XXX function pointer types in pkcs11.h:
+ *
+ * 'dllimport' attribute only applies to functions, variables, classes, and Objective-C interfaces
+ *
+ * I'm not convinced the dllimport is needed/approproate for typedefs under MSVC either.
+*/
+#if 0  // defined(_MSC_VER)
    #define CK_DECLARE_FUNCTION_POINTER(returnType, name) returnType __declspec(dllimport)(*name)
 #else
    #define CK_DECLARE_FUNCTION_POINTER(returnType, name) returnType(*name)
```

Comment:
I'm not too sure either, but I am assuming any breakage will be really obvious linker or loader failures rather than anything subtle.

Reply:
Fair. And it does sound somewhat odd to "dllimport" a function pointer declaration (assuming it is not a global variable or static member variable that is defined in some dllexporting compilation unit). For the sake of caution, we could also special-case this for clang-cl, I guess:

```suggestion
// When building with clang-cl on Windows, specifying this with dllimport causes
// a warning when defining the CK_C_XXX function pointer types in pkcs11.h:
//
//   'dllimport' attribute only applies to functions, variables, classes, and Objective-C interfaces
//
// ... hence, we enable this for MSVC only.
#if defined(BOTAN_BUILD_COMPILER_IS_MSVC)
   #define CK_DECLARE_FUNCTION_POINTER(returnType, name) returnType __declspec(dllimport)(*name)
#else
   #define CK_DECLARE_FUNCTION_POINTER(returnType, name) returnType(*name)
```

@randombit, do you have an opinion here? Also regarding semver, perhaps?

Otherwise: if we all agree with not dllimport-declaring this any longer, I'd say we should get rid of the `#if 0` and just 
 leave the bare `#define` from the else-branch.

Reply:
Even the latest (that I can find) header in https://docs.oasis-open.org/pkcs11/pkcs11-base/v3.0/os/include/pkcs11-v3.0/pkcs11.h has similar commentary. That said I agree it doesn't seem to make sense to "dllimport" this, *and* it does seem like if it is incorrect it would probably fail in an obvious (eg link error) vs subtle way.

I don't see any SemVer implications to this either way.

I guess I'd be ok with either leaving it as MSVC specific or just removing it entirely and using the `#else` branch everywhere.